api/lib/read-graphs/attribute.ts

```diff
@@ -51,26 +41,22 @@ export async function getTableAttributes (tableId: string) {
     .prefixes({ hydra, dataCube, rdf, api })
     .execute(getClient()))
 
-  await collection.import(await construct()
-    .graph(`
-      ?attribute dataCube:columnMapping [
-        dataCube:sourceColumn ?sourceColumn ;
-        dataCube:referencedColumn ?referencedColumn ;
-      ] .
-
-      ?sourceColumn schema:name ?sourceName .
-      ?referencedColumn schema:name ?referencedName .`)
+  await collection.import(await describe('?attribute', '?params', '?mapping')
     .where(`
       ?attribute dataCube:table <${tableId}> .
-      ?attribute dataCube:columnMapping [
-        dataCube:sourceColumn ?sourceColumn ;
-        dataCube:referencedColumn ?referencedColumn ;
-      ] .
+      ?attribute a dataCube:Attribute .
 
-      ?sourceColumn schema:name ?sourceName .
-      ?referencedColumn schema:name ?referencedName .
+      OPTIONAL {
+        ?attribute dataCube:columnMapping ?mapping .
+      }
+
+      OPTIONAL {
+        ?attribute <https://rdf-cube-curation.described.at/datatype/parameters> ?params .
```

Comment:
Could you also use the dataCube prefix here?

Reply:
Yea, unfortunately not. `dataCube:datatype/parameters` is not a valid prefixed name.

Though I probably should add the `datatype` prefix which I used elsewhere